opencv/src/OpenCV/ImgProc/StructuralAnalysis.hsc

```diff
@@ -399,3 +400,24 @@ minAreaRect points =
       |]
   where
     c'numPoints = fromIntegral $ V.length points
+
+{- |  Tests a contour convexity.
+
+The function tests whether the input contour is convex or not.
+The contour must be simple, that is, without self-intersections.
+Otherwise, the function output is undefined.
+
+<http://docs.opencv.org/3.0-last-rst/modules/imgproc/doc/structural_analysis_and_shape_descriptors.html?highlight=contourarea#iscontourconvex OpenCV Sphinx doc>
+-}
+isContourConvex
+    :: (IsPoint2 point2 CFloat)
+    => V.Vector (point2 CFloat)
+    -> Bool
+isContourConvex contour = toBool $ unsafePerformIO $
+    withArrayPtr (V.map toPoint contour) $ \contourPtr ->
+        [CU.exp| bool {
```

Comment:
The c++ implementation of `cv::isContourConvex` [throws exceptions](https://github.com/opencv/opencv/blob/master/modules/imgproc/src/convhull.cpp#L391).

So you must use the `cvExcept` quasi quoter instead of `CU.exp`.

That also means you have to manually `alloca` a `Ptr` to `Bool` to store the result.